src/sentry/api/bases/organization_events.py

```diff
@@ -312,42 +313,71 @@ def get_event_stats_data(
 
         with sentry_sdk.start_span(op="discover.endpoint", description="base.stats_serialization"):
             # When the request is for top_events, result can be a SnubaTSResult in the event that
             # there were no top events found. In this case, result contains a zerofilled series
             # that acts as a placeholder.
+            is_multiple_axis = len(query_columns) > 1
             if top_events > 0 and isinstance(result, dict):
                 results = {}
                 for key, event_result in result.items():
-                    if len(query_columns) > 1:
+                    if is_multiple_axis:
                         results[key] = self.serialize_multiple_axis(
-                            serializer, event_result, columns, query_columns, allow_partial_buckets
+                            serializer,
+                            event_result,
+                            columns,
+                            query_columns,
+                            allow_partial_buckets,
+                            zerofill_results=zerofill_results,
                         )
                     else:
                         # Need to get function alias if count is a field, but not the axis
                         results[key] = serializer.serialize(
                             event_result,
                             column=resolve_axis_column(query_columns[0]),
                             allow_partial_buckets=allow_partial_buckets,
+                            zerofill_results=zerofill_results,
                         )
-                return results
-            elif len(query_columns) > 1:
-                return self.serialize_multiple_axis(
-                    serializer, result, columns, query_columns, allow_partial_buckets
+                serializedResult = results
+            elif is_multiple_axis:
+                serializedResult = self.serialize_multiple_axis(
+                    serializer,
+                    result,
+                    columns,
+                    query_columns,
+                    allow_partial_buckets,
+                    zerofill_results=zerofill_results,
                 )
             else:
-                return serializer.serialize(
+                serializedResult = serializer.serialize(
                     result,
                     resolve_axis_column(query_columns[0]),
                     allow_partial_buckets=allow_partial_buckets,
+                    zerofill_results=zerofill_results,
                 )
+            if hasattr(result, "start") and hasattr(result, "end"):
+                if is_multiple_axis:
+                    for value in serializedResult.values():
+                        value["start"] = result.start
+                        value["end"] = result.end
+                else:
+                    serializedResult["start"] = result.start
+                    serializedResult["end"] = result.end
```

Comment:
I might have missed convo on this already, but event-stats only takes one start&end value overall.
Is this for statsPeriods where we send relative ranges, and if so could we just calculate the range on the frontend instead?

Reply:
Hey Will, for queries with ranges selected such as last 24hrs, 7days, etc, the frontend doesn't actually send the time range, just the `statPeriod`. From what I can tell, it seems like the backend is responsible for determining the exact time ranges based on the supplied `statPeriod`. Because of this, if we were to calculate the time range on the frontend, it might not match the exact time range of the result from the backend. The time difference would probably not have noticeable impact to the experience but I feel like it might be more clean to get the exact range from the backend despite the increase in complexity. Let me know what you think or if I've gotten something wrong here!